### ISSUE TITLE
[JAR Task] Java heap space

### ISSUE THREAD

**Issue by dleangen** [bndtools/bnd#3875]
When running ```gradle clean build``` using ```bnd_version=5.0.1``` I get this:
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':com.foo.application:jar'.
> Java heap space
```

I have been trying and trying to understand the pattern, but without success.

This is the bnd.bnd file:

```
# Do not edit this file.

# Please read the "readme.md" file for instructions.

-dependson: 			*

-export: 				application.bndrun

-fixupmessages: "No translation found", "Unused Private-Package"

-baseline: 
-baselinerepo: 
-releaserepo: 
-includeresource: 
```

This is the application.bndrun file:

```
-include: bnd.bnd

-runsystempackages.sws:	sun.security.ssl

-runrequires: \
	bnd.identity;id=...

-runbundles: \
	...
```

The closest I have come to understanding a "pattern" (but without much consistency) is that if the total size of the application jar is >30M or so, the error occurs. This seems like a very small limit for an application jar!!

**Comment by dleangen** [comment]
I noticed [this issue](https://github.com/bndtools/bnd/issues/3400), but I do not use CDI, and I do not have a "Gigabit" project. As I mentioned, the total is only ~35M or so.

I also tested each project one-by-one to ensure there wasn't some kind of corrupt file in any of the projects. Each project individually seems to work just fine.

**Comment by bjhargrave** [comment]
Did you try setting a larger heap for the gradle build? We have to do so for the Bnd build since the default is not enough.

https://github.com/bndtools/bnd/blob/1ed18d5e51679dcfcd561abe2fe50f76120640a4/gradle.properties#L1-L2

**Comment by dleangen** [comment]
Hmmmm. Setting a larger heap was indeed the first thing I thought about and took action on. I set ```GRADLE_OPTS``` to exactly the same values you provided.

It appears that setting ```GRADLE_OPS``` as an environment variable and ```org.gradle.jvmars``` in ```gradle.properties``` does not behave in the same way. When I set ```GRADLE_OPS```, my build still fails. However, when I set the property in ```gradle.properties```, it is now working.

Sorry. I assumed it would have the same effect. It is not at all intuitive that it would be different.
